modules/substrate/src/lib.rs

```diff
@@ -371,15 +373,79 @@ impl<T: Config> bp_header_chain::HeaderChain<BridgedHeader<T>> for Module<T> {
 		PalletStorage::<T>::new().current_authority_set()
 	}
 
-	fn import_header(header: BridgedHeader<T>) -> Result<(), ()> {
-		let mut verifier = verifier::Verifier {
-			storage: PalletStorage::<T>::new(),
-		};
+	fn append_finalized_chain(headers: impl IntoIterator<Item = BridgedHeader<T>>) {
+		let mut storage = PalletStorage::<T>::new();
 
-		let _ = verifier.import_header_unchecked(header).map_err(|_| ())?;
+		for header in headers.into_iter() {
+			import_header_unchecked::<_, T>(&mut storage, header);
+		}
+	}
+}
 
-		Ok(())
+/// Import a finalized header without checking if this is true.
+///
+/// This function assumes that all the given header has already been proven to be valid and
+/// finalized. Using this assumption it will write them to storage with minimal checks. That
```

Comment:
```suggestion
/// finalized, and a child of current best finalized header. Using this assumption it will write them to storage with minimal checks. That
```